### PR TITLE
[ci-skip] Add 6.0 Release Notes for Active Record

### PR DESCRIPTION
**Pull request by alexcameron89**
### Summary

This PR adds Rails 6.0 Release Notes to Active Record based off of the Changelog.

### Thoughts related to this PR

Not knowing what specifically should be notable changes, I opted for including more than less. I'm glad to remove any that should not be included.

I also thought that it would be nice to categorize the notable changes with headers based off of major features, version bumps, or fixes. While I haven't implemented that change, I'm very glad to do so.

### CODE REVIEW

File: guides/source/6_0_release_notes.md [hunk 1]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+    ([Commit](https://github.com/rails/rails/commit/9def05385f1cfa41924bb93daa187615e88c95b9))
+
+*   Deprecate using class level querying methods if the receiver scope has leaked.
+    ([Commit](https://github.com/rails/rails/commit/4c6171d6056a346a953792a954853d14c000dc90))
+
+*   Deprecate `config.activerecord.sqlite3.represent_boolean_as_integer`.
+    ([Commit](https://github.com/rails/rails/commit/f59b08119bc0c01a00561d38279b124abc82561b))
+
+*   Deprecate passing `migrations_paths` to `connection.assume_migrated_upto_version`.
+    ([Commit[(https://github.com/rails/rails/commit/c1b14aded27e063ead32fa911aa53163d7cfc21a))
```
Comment (kamipo) [inline guides/source/6_0_release_notes.md]:
```suggestion
    ([Commit](https://github.com/rails/rails/commit/c1b14aded27e063ead32fa911aa53163d7cfc21a))
```

File: guides/source/6_0_release_notes.md [hunk 2]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+    ([Commit](https://github.com/rails/rails/commit/27b252d6a85e300c7236d034d55ec8e44f57a83e))
+
+
 ### Deprecations
 
+*   Deprecate mismatched case-sensitivity collation comparisons for uniqueness validator.
+    ([Commit](https://github.com/rails/rails/commit/9def05385f1cfa41924bb93daa187615e88c95b9))
+
+*   Deprecate using class level querying methods if the receiver scope has leaked.
+    ([Commit](https://github.com/rails/rails/commit/4c6171d6056a346a953792a954853d14c000dc90))
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: Can we link to the PR https://github.com/rails/rails/pull/35280 here?

File: guides/source/6_0_release_notes.md [hunk 3]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+    `columns_per_table`, `indexes_per_table`, `columns_per_multicolumn_index`,
+    `sql_query_length`, and `joins_per_query`.
+    ([Commit](https://github.com/rails/rails/commit/e0a1235f7df0fa193c7e299a5adee88db246b44f))
+
+*   Deprecate `update_attributes`/`!` in favor of `update`/`!`.
+    ([Commit](https://github.com/rails/rails/commit/5645149d3a27054450bd1130ff5715504638a5f5))
+
 ### Notable changes
 
+*   Add `rails db:prepare` to migrate or setup a database by creates the database, loading the schema,
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: We don't need `by creates the database` right?

File: guides/source/6_0_release_notes.md [hunk 4]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+*   Add `#destroy_by` and `#delete_by` for conditional removals.
+    ([Pull Request](https://github.com/rails/rails/pull/35316))
+
+*   Add the ability to automatically switch database connections.
+    ([Pull Request](https://github.com/rails/rails/pull/35073))
+
+*   Add the ability to prevent writes to a database for the duration of a block.
+    ([Pull Request](https://github.com/rails/rails/pull/34505))
+
+*   Add an API for switching connections to support multiple databases. This
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: Perhaps we can just keep one line. `Add an API for switching connections to support multiple databases.`

File: guides/source/6_0_release_notes.md [hunk 5]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+    ([Pull Request](https://github.com/rails/rails/pull/33378))
+
+*   Bump the minimum MySQL version to 5.5.8.
+    ([Pull Request](https://github.com/rails/rails/pull/33853))
+
+*   Use the utf8mb4 character set by default in MySQL.
+    ([Pull Request](https://github.com/rails/rails/pull/33608))
+
+*   Add the ability to filter out sensitive data in `#inspect`
+    (Pull Requests: [One](https://github.com/rails/rails/pull/33756) & [Two](https://github.com/rails/rails/pull/34208))
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: We are following this style for multi PRs https://monosnap.com/file/N7mfqH7seXlyzhbru5JX4moxRtLan2

File: guides/source/6_0_release_notes.md [hunk 6]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+
+*   Change `ActiveRecord::Base.configurations` to now return an object instead of a hash.
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: `now` can be dropped.

File: guides/source/6_0_release_notes.md [hunk 7]
```diff
@@ -211,10 +211,240 @@ Please refer to the [Changelog][active-record] for detailed changes.
...
+*   Add `touch_all` method to `ActiveRecord::Relation`.
+    ([Pull Request](https://github.com/rails/rails/pull/31513))
+
+*   Add `ActiveRecord::Base.base_class?` predicate.
+    ([Pull Request](https://github.com/rails/rails/pull/32417))
+
+*   Add custom prefix/suffix options to `ActiveRecord::Store.store_accessor`.
+    ([Pull Request](https://github.com/rails/rails/pull/32306))
+
+*   Rails 6 requires Ruby 2.5.0 or newer.
```
Comment (prathamesh-sonpatki) [inline guides/source/6_0_release_notes.md]: We can drop this entry. It is present in other sections.

**Review by prathamesh-sonpatki — CHANGES_REQUESTED** [review]

**Comment by simi** [comment]
I think https://github.com/rails/rails/pull/35844 is missing.

**Comment by alexcameron89** [comment]
@prathamesh-sonpatki thank you very much for the thorough review! I've updated my PR to reflect the changes.

@simi Question: is the noteworthy change in #35844 that Rails 6 is compatible with `sqlite3 1.4.0`?

File: guides/source/6_0_release_notes.md (hunk 3 above)
Comment (alexcameron89) [reply guides/source/6_0_release_notes.md]: I re-reviewed the PR and I've updated the description to:

> Add `rails db:prepare` to create a database if it doesn't exist, and run its migrations.

Does that sound right to you?

**Comment by simi** [comment]
@alexcameron89 it is not making AR compatible with 1.4, but actually it makes AR incompatible with `< 1.4`. That's a huge change IMHO.

**Comment by alexcameron89** [comment]
@simi That makes sense, thank you for the additional context. I added it in 029da21

**Review by prathamesh-sonpatki — APPROVED** [review]

**Comment by prathamesh-sonpatki** [comment]
@alexcameron89 Can you squash the commits into one?

**Review by simi — APPROVED** [review]

**Comment by alexcameron89** [comment]
@prathamesh-sonpatki done 👍

**Pull request merged**